### ISSUE TITLE
listeningTo the same event on different models is broken (events.split /  cannot read property 'obj' of undefined / etc.)

### ISSUE THREAD

**Issue by supernintendo** [spine/spine#603]
With `Spine.Controller`, if you `listenTo` or `listenToOnce` an event on a model when the controller is already listening to the same event on a different model, the associated object in the `listeningTo` / `listeningToOnce` array will contain an incorrect reference to the model of the first listener. The side effect of this is that `stopListeningTo` breaks because the controller is trying to remove the same listeners more than once for the same model.

The following PR intends to fix this bug but really just hides it: https://github.com/spine/spine/commit/fd958b6098412844c0421f1928fdcf42bf74bc54

Suggestion: Fix `listenTo` and `listenToOnce` so that the objects it uses to allow `stopListening` to work contain the correct properties. A nice improvement would also be to add an `isListeningTo` function to allow for conditionally removing listeners.

I'll look into what is causing this issue and submit a PR.

**Comment by aeischeid** [comment]
`isListeningTo` sounds like an okay idea.

any PRs will be welcome. sounds like you have a good enough idea of what is going on to write a failing test at least? or maybe patch up one of the tests that is falsely passing?

**Comment by supernintendo** [comment]
@aeischeid Yeah, I'll certainly include a test so this doesn't regress. I'll have some time over the weekend to make this fix. Thanks for the quick response!

**Comment by wschmrdrNS** [comment]
While said devs toil especially hard on this, I ran into this as well, and did find that I was able to use bind, one, and unbind as a workaround for the time being.
